### DISCUSSION TITLE
posts/rancher-ha-install/

### DISCUSSION THREAD

**Discussion by giscus** [techno-tim/techno-tim.github.io#227] [Comments]
# posts/rancher-ha-install/

Documentation for Techno Tim YouTube videos for all your copy pasta needs!

https://technotim.live/posts/rancher-ha-install/

**Comment by niinod** [comment]
First of all. Thanks for all the awesome videos! But I have a problem.

I followed this to get rancher working, after setting up my cluster from your k3s + ansible video. 
Now, after installing rancher, and pointing rancher.lan.mydomain.io to the ip of the nginx welcome page, I still just reach the nginx welcome page. (Maybe I need to edit the nginx config? I can't find it.)

Im not sure what I need to do, to get it to redirect to rancher? is there some detail that I've missed or have I misunderstood something?

**Reply by properlyabsurd** [reply]
Tim included a link on how to configure the nginx load balancer.

https://rancher.com/docs/rancher/v2.x/en/installation/resources/k8s-tutorials/infrastructure-tutorials/nginx/

I use Traefik, so I can't give any pointers on nginx.

**Reply by xXAzazelXx** [reply]
Hey mate,

Any change you can please explain how you did that on Traefik?
I've installed k3s + ansible + Traefik as oer Tim's videos, and now im at the LoadBalancer part

**Comment by Dodoooh** [comment]
I am new to the K3S world, I created it with the awesome ansible script. 

Now I have installed the cert-manager with traefik as well as Rancher. 
https://technotim.live/posts/kube-traefik-cert-manager-le
https://technotim.live/posts/rancher-ha-install

But now I have the question, how can I give rancher a let's encrypt tls certificate? With the example for NGNIX it worked fine, but rancher is installed differently, so I don't really understand this. 

Thanks a lot

**Comment by pabloaugusto** [comment]
And we who choice Kube-VIP as LB at ks3 config yaml?
Whats the best way to catch the IP from KUBE-VIP LB?

**Comment by xXAzazelXx** [comment]
Thank you for this!
Has anyone managed to get Rancher to work on the latest k3s folling ansible deploy?
I think th elatest kube is too new

Error: INSTALLATION FAILED: chart requires kubeVersion: < 1.28.0-0 which is incompatible with Kubernetes v1.29.2+k3s1

**Reply by pabloaugusto** [reply]
Max k3s supported version for rancher, at this momment, is `v1.27` https://www.suse.com/suse-rancher/support-matrix/all-supported-versions/rancher-v2-8-2/ change your install settings to v1.27+k3s1 in order to work with rancher ;)

**Reply by xXAzazelXx** [reply]
Thank you!

**Comment by guyein** [comment]
Followed carefully but for some reason when got to `install cert-manager helm chart` step, I get: 

`Error: INSTALLATION FAILED: cannot re-use a name that is still in use`

**Reply by jyyyeung** [reply]
Try replacing `install` with `upgrade`

**Comment by pabloaugusto** [comment]
> Error: INSTALLATION FAILED: cannot re-use a name that is still in use

Google + Stackoverflow S2

https://stackoverflow.com/questions/70464815/cannot-install-kubernetes-helm-chart-error-cannot-re-use-a-name-that-is-still-i

**Comment by thenbirdpan** [comment]
I am currently hitting a issue were Rancher is not getting a external ip.
I have followed all the steps provided.
Does anyone have any idea what I could be doing wrong?

Note I followed the HA tutorial and only changing the k3s version, and this a Pi4 Cluster


kubectl -n cattle-system rollout status deploy/rancher
deployment "rancher" successfully rolled out

kubectl get nodes
NAME            STATUS   ROLES                                       AGE   VERSION
kube-master-1   Ready    control-plane,etcd,master   40m   v1.28.10+k3s1
kube-master-2   Ready    control-plane,etcd,master   39m   v1.28.10+k3s1
kube-worker-2   Ready    <none>                                   38m   v1.28.10+k3s1

     
kubectl get svc --all-namespaces -o wide              
NAMESPACE                                    NAME                   EXTERNAL-IP   
cattle-fleet-system                          gitjob                 "none"        
cattle-provisioning-capi-system              capi-webhook-service   "none"        
cattle-system                                rancher                "none"        
cattle-system                                rancher-webhook        "none"        
cert-manager                                 cert-manager           "none"        
cert-manager                                 cert-manager-webhook   "none"        
default                                      kubernetes             "none"        
kube-system                                  kube-dns               "none"        
kube-system                                  metrics-server         "none"        
metallb-system                               webhook-service        "none"
